### ISSUE TITLE
Adding querystring to controller/action instead of just the slug

### ISSUE THREAD

**Issue by MCNorthwood** [PiranhaCMS/piranha.core#362]
Version 5.0.0
An issue that has happened on occasion is adding the guid onto the link such as this:
**/home?startpage=gehr&piranha_handled=gehr&id=qpp90q29-769270-440475-8n77-r8211574028onq&id=dbc90d29-7692-4404-8a77-e82115741bad&startpage=true&piranha_handled=true**

In this instance dbc90d29-7692-4404-8a77-e82115741 is the guid for the homepage, however seems to also be looking up a separate guid first and creates a bad request? This creates a 500 error with a NullReferenceException for the home/Index, which assume is due to the first guid in the querystring that is coming from seemingly nowhere. I checked the PageFields table also to see if it's in there it ain't, it's not the siteId either.
Not sure whether this is a problem that is resolved already in a later version or not?

This is only come to attention due to having _application insights_ and it showing a 404 because it also did it with the route of the page with the controller/action then adding the querystring also.
so **/foo/bar?id=d4be2c3e-cbb1-408f-aa48-12d7bff9bd7c&startpage=false&piranha_handled=true** instead of the slug that was set, which was just /bar. It however would go to the page and have that as the slug and use the controller and action instead of the slug, but come up as a 404 but still showed the correct page.

tl:dr One ends up causing a 500, the other causes a 404 because it passes through the correct id, however doesn't seem to return the correct slug.

**Comment by tidyui** [comment]
Well I’m curious to how the first url was created, the second is however completely valid from an internal perspective as this is how the routing works.

When a url is intercepted the middleware tries to match it against a slug. If it finds it it **rewrites** the request internally to the route handling the content type and then adds query string params for:

* the content Id 
* if it is the site start page 

It also adds a param telling the remaining middleware that the request has been handled so the other middleware don’t try to resolve the request.

The first request however looks like it’s been rewritten **twice**, also I can’t see where the “gehr”, “onq” and “bad” should come from, is this an external requested url?

Regards

**Comment by tidyui** [comment]
And for clarification, the slug is **never** passed on to the controller action from the middleware, it’s always the parameters you showed above. You can take a look at the core project **Piranha.AspNetCore** to look at how the middleware works.

Also, if you need your middleware to work differently for some reason you can just write your own and rewrite the requests any way you like, everything is pluggable!

**Comment by MCNorthwood** [comment]
So testing a little more. I have no clue how anyone ends up getting the querystring when accessing the site.
However when I'm looking more into it. There's nothing that seems to change it if the querystring is put in explicitly. So the permalink, although is returning the correct permalink from whichever was put in the slug for the DynamicPage that we have. 

Midway through this I've figured out what's happening. It's because our slug and the controller has the same name of home. So with the added querystring it's simply thinking it's the correct link and doesn't correct the permalink to just /home.

Thanks for your help but it's no longer an issue :)

**Comment by tidyui** [comment]
Aha. I'm glad you figured it out!
